### ISSUE TITLE
Loading missing files on localhost does not throw 404

### ISSUE THREAD

**Issue by TimvdLippe** [PolymerLabs/polyserve#80]
When you open the polyserved pages on `localhost:8080/` invalid urls (404's) are not thrown. Instead `index.html` is served. Opening `http://localhost:8080/components/componentname` does throw the 404 as expected.

## Steps to reproduce
Index.html:
```html
<html>
<head>
  <link rel="import" href="incorrect.html">
</head>
</html>
```

1. Run `polyserve`
* Open `localhost:8080/`
* Open dev tools
* Observe that the HTML import resolved to `index.html`  instead of throwing a 404

**Comment by justinfagnani** [comment]
This is intended behavior. We're setting polyserve to work for apps that use client-side routing and serve up a shell. The pattern we're implementing in the dev server is to serve up the main entrypoint instead of 404'ing, and then the shell can handle the route as needed. We'll be documenting this behavior more very soon.

We're going to refine the behavior a lot in the next couple of weeks. At least we need to exclude bower_components/, components/ and dotfiles. I'm very curious where it might not be appropriate. Do you have a particular case where it's a problem?

Sorry for the disruption.

**Comment by TimvdLippe** [comment]
Thank you for the explanation. I do see the benefit of catching 404's in the application, however when loading files this is counterintuitive. We have spent considerable amount of time debugging until we discovered it failed silently.

Therefore I propose to keep the old behavior as default, with an option to turn the silent failures on.
It might also be possible to silently catch all 404's on regular paths (e.g. `/foo/bar`) while throwing 404's on paths leading to files (e.g. `/foo/bar.html` or `/foo/bar.js`). WDYT?

**Comment by TimvdLippe** [comment]
Right now I am encountering issues with not found files. Since `polyserve` instead loads `index.html`, it starts to continuously load all files again. This results in an infinite loop of loading and failing. Debugging this issue is near impossible.

**Comment by TimvdLippe** [comment]
For reference: this issue has actually been fixed in #90
